RELEASING.md

```diff
@@ -0,0 +1,72 @@
+How To Release
+==============
+
+Due to Maven Central's very particular requirements, the release process is a bit
+elaborate and requires a good deal of local configuration.  This guide should walk
+you through it.  It won't do anyone outside of KeepSafe any good, but the workflow
+is representative of just about any project deploying via Sonatype.
+
+We currently deploy to Maven Central (via Sonatype's OSS Nexus instance).
+
+### Prerequisites
+
+1. A *published* GPG code-signing key
+1. A Sonatype Nexus OSS account with permission to publish in com.getkeepsafe
+1. A plugins.gradle.org account with permission to publish in com.getkeepsafe
+1. Permission to push directly to https://github.com/KeepSafe/ReLinker
+
+### Setup
+
+1. Add your GPG key to your github profile - this is required
+   for github to know that your commits and tags are "verified".
+1. Configure your code-signing key in ~/.gradle.properties:
+   ```gradle
+   signing.keyId=<key ID of your GPG signing key>
+   signing.password=<your key's passphrase>
+   signing.secretKeyRingFile=/path/to/your/secring.gpg
+   ```
+1. Configure your Sonatype credentials in ~/.gradle.properties:
+   ```gradle
+   SONATYPE_NEXUS_USERNAME=<nexus username>
+   SONATYPE_NEXUS_PASSWORD=<nexus password>
+   ```
+1. Configure git with your codesigning key; make sure it's the same as the one
+   you use to sign binaries (i.e. it's the same one you added to gradle.properties):
+   ```bash
+   # Do this for the ReLinker repo only
+   git config user.email "your@email.com"
+   git config user.signingKey "your-key-id"
+   ```
+
+### Pushing a build
+
+1. Edit gradle.properties, update the VERSION property for the new version release
+1. Edit readme so that Gradle examples point to the new version
+1. Edit changelog, add relevant changes, note the date and new version (follow the existing pattern)
+1. Add new `## [Unreleased]` header for next release
+1. Verify that the everything works:
+   ```bash
+   ./gradlew clean check
+   ```
+1. Make a *signed* commit:
+   ```bash
+   git commit -S -m "Release version X.Y.Z"
+   ```
+1. Make a *signed* tag ()check existing tags for message format):
+   ```bash
+   git tag -S -a X.Y.Z
+   ```
+1. Upload binaries to Sonatype:
+   ```bash
+   ./gradlew publish
```

Comment:
I think the command should be `./gradlew uploadArchives`?

Reply:
I think the `./gradlew uploadArchives` task comes from the jcenter gradle plugin `'com.jfrog.bintray'`.

`com.vanniktech.maven.publish` provides the following tasks:
```
Publishing tasks
----------------
generateMetadataFileForMavenPublication - Generates the Gradle metadata file for publication 'maven'.
generatePomFileForMavenPublication - Generates the Maven POM file for publication 'maven'.
publish - Publishes all publications produced by this project.
publishAllPublicationsToLocalRepository - Publishes all Maven publications produced by this project to the local repository.
publishAllPublicationsToMavenRepository - Publishes all Maven publications produced by this project to the maven repository.
publishMavenPublicationToLocalRepository - Publishes Maven publication 'maven' to Maven repository 'local'.
publishMavenPublicationToMavenLocal - Publishes Maven publication 'maven' to the local Maven repository.
publishMavenPublicationToMavenRepository - Publishes Maven publication 'maven' to Maven repository 'maven'.
publishToMavenLocal - Publishes all Maven publications produced by this project to the local Maven cache.

Release tasks
-------------
closeAndReleaseRepository - Closes and releases an artifacts repository in Nexus
```

Reply:
Interesting.  I use `uploadArchives` regularly.  If `publish` works here, then it's fine to leave it.  I was more worried that it was a legacy of the old upload stuff.